common/server_protocolhandler.cpp

```diff
@@ -627,16 +627,45 @@ Response::ResponseCode Server_ProtocolHandler::cmdJoinRoom(const Command_JoinRoo
                 return Response::RespUserLevelTooLow;
         }
 
-        if (roomPermission == "moderator"){
+        if (roomPermission == "moderator") {
```

Comment:
Can you encapsulate every roomPermission check into its own function?

```cpp
bool userIsPermittedToJoin(QString roomPermission, ServerInfo_User userInfo)
```

Then inside this function we can just
```cpp
if (!userIfPermittedToJoin(roomPermission, userInfo)
    return Response::RespUserLevelTooLow

```

If possible I think that function could be static - this would make it easier to unit test later